### PR TITLE
Add human phenotype ontology

### LINKED ISSUE #170: Add HPO
https://hpo.jax.org/ & https://hpo.jax.org/app/data/ontology

Probably under it's own entity.

### PR DESCRIPTION
**Pull request by Zethson**
Signed-off-by: zethson <lukas.heumos@posteo.net>

### CODE REVIEW

File: bionty/_table.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def _camel_to_snake(string: str) -> str:
 class EntityTable:
     """Biological entity as a EntityTable.
 
-    See :doc:`guide/index` for background.
+    See :doc:`guide/lookup` for background.
```
Comment (Zethson) [inline bionty/_table.py]: Led to a documentation error. Think this is the correct page to point to?

File: bionty/versions/versions.yaml [hunk 2]
```diff
@@ -32,3 +32,7 @@ Readout:
   efo:
     versions:
       3.48.0: http://www.ebi.ac.uk/efo/releases/v3.48.0/efo.owl
+Phenotype:
+  hpo:
+    versions:
+      2022-01-27: http://purl.obolibrary.org/obo/hp.owl
```
Comment (Zethson) [inline bionty/versions/versions.yaml]: I could not find a stable, versionized source yet. In other words, this will always point to the latest release.

**Comment by Zethson** [comment]
Think the issue for the 3.10 CI is that it searches for `"_current.yaml"` for the version which doesn't yet include the phenotype?

Shouldn't this get fixed by `_sync_versions`?

File: bionty/versions/versions.yaml (hunk 2 above)
Comment (Zethson) [reply bionty/versions/versions.yaml]: Possible to get them from the GH releases -> https://github.com/obophenotype/human-phenotype-ontology/releases I shall fix the URL

File: bionty/_table.py (hunk 1 above)
Comment (sunnyosun) [reply bionty/_table.py]: Hmm, pointing to `guide/index` should work, and it's working in the API page on the current deployment: https://lamin.ai/docs/bionty/bionty.entitytable#bionty.EntityTable

What error did you see?

File: bionty/versions/versions.yaml (hunk 2 above)
Comment (sunnyosun) [reply bionty/versions/versions.yaml]: Right, the stable link seems to have not been generated yet, I only saw the 2022 version here: https://www.ebi.ac.uk/ols/ontologies/hp

FYI, `bt.dev.ontology_info("hp")`looks into https://www.ebi.ac.uk/ols/ontologies/ and finds most recent stable version link in ebi.

But in this case, yes we need to use the github release linked file if we want 2023.

File: bionty/_phenotype/_core.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+
+class Phenotype(EntityTable):
+    """Phenotype.
+
+    Edits of terms are coordinated and reviewed on:
+    https://hpo.jax.org/app/
+    """
+
+    def __init__(
```
Comment (sunnyosun) [inline bionty/_phenotype/_core.py:25]: could you add a species parameter here (similar to https://github.com/laminlabs/bionty/blob/main/bionty/_cellmarker/_core.py#L23) and defaults to human? There is also mammalian phenotype ontology (https://www.ebi.ac.uk/ols/ontologies/mp) and potentially others, we should be specific here.

File: bionty/versions/versions.yaml [hunk 4]
```diff
@@ -32,3 +32,7 @@ Readout:
   efo:
     versions:
       3.48.0: http://www.ebi.ac.uk/efo/releases/v3.48.0/efo.owl
+Phenotype:
+  hpo:
```
Comment (sunnyosun) [inline bionty/versions/versions.yaml]: Is this hp or hpo? The url is hp, but I saw description uses hpo. In general the naming seems a bit mixed up, such as efo, clo uses with o in their url, but cl doesn't. 

I'd suggest to keep the same as what's in the url, in this case hp, what do you think?

**Review by sunnyosun — APPROVED** [review]
Looks great, thank you! 👍

File: bionty/versions/versions.yaml (hunk 4 above)
Comment (Zethson) [reply bionty/versions/versions.yaml]: I don't have a strong opinion on this so I'll go with your suggested `hp`

File: bionty/_table.py (hunk 1 above)
Comment (Zethson) [reply bionty/_table.py]: A file not found error. But I reverted it for now and will keep an eye on it.

**Pull request merged**